src/status_im/bots/events.cljs

```diff
@@ -52,25 +56,40 @@
 
 (def ^:private keywordize-vector (partial mapv keyword))
 
-;; TODO(janherich): do something with this horrible triple nested reduce so it's more readable
+(defn- transform-bot-subscriptions
+  "Transforms bot subscriptions as returned from jail in the following format:
+
+  `{:calculatedFee {:subscriptions {:value [\"sliderValue\"]
+                                    :tx [\"transaction\"]}}
+    :feeExplanation {:subscriptions {:value [\"sliderValue\"]}}}`
+
+  into data-structure better suited for subscription lookups based on changes
+  in `:bot-db`:
+
+  `{[:sliderValue] {:calculatedFee {:value [:sliderValue]
+                                    :tx [:transaction]}}
+    [:transaction] {:feeExplanation {:value [:sliderValue]}}}`
+
+  In the resulting data-structure, the top level keys are the (keywordized) paths
+  in the `:bot-db` data structure, so it's quick and easy to look-up all the
+  subscriptions which must be recalculated when something in their path changes."
+  [bot-subscriptions]
```

Comment:
Looks like example is broken:
```clojure
(transform-bot-subscriptions
 {:calculatedFee  {:subscriptions {:value ["sliderValue"]
                                   :tx    ["transaction"]}}
  :feeExplanation {:subscriptions {:value ["sliderValue"]}}})

=>
{[:sliderValue] {:calculatedFee  {:value [:sliderValue], :tx [:transaction]},
                 :feeExplanation {:value [:sliderValue]}},
 [:transaction] {:calculatedFee {:value [:sliderValue], :tx [:transaction]}}}
```

Reply:
Sorry, that's my mistake in the docstring, I will correct it, thanks for pointing it out !